src/solvers/solver-HQP-eiquadprog-fast.cpp

```diff
@@ -86,105 +87,112 @@ namespace tsid
+                i_in += constr->rows();
+                m_qpData.CI.middleRows(i_in, constr->rows()) = -constr->matrix();
+                m_qpData.ci0.segment(i_in, constr->rows())   = constr->upperBound();
+                i_in += constr->rows();
+              }
+              else if(constr->isBound())
+              {
+                m_qpData.CI.middleRows(i_in, constr->rows()).setIdentity();
+                m_qpData.ci0.segment(i_in, constr->rows())   = -constr->lowerBound();
+                i_in += constr->rows();
+                m_qpData.CI.middleRows(i_in, constr->rows()) = -Matrix::Identity(m_n, m_n);
+                m_qpData.ci0.segment(i_in, constr->rows())   = constr->upperBound();
+                i_in += constr->rows();
+              }
+            }
           }
-        }
-      }
-      else
-        resize(m_n, neq, nin);
+          else
+            resize(m_n, neq, nin);
       
-      EIGEN_MALLOC_NOT_ALLOWED;
-
-      if(problemData.size()>1)
-      {
-        const ConstraintLevel & cl1 = problemData[1];
-        m_H.setZero();
-        m_g.setZero();
-        
-        
-        for(ConstraintLevel::const_iterator it=cl1.begin(); it!=cl1.end(); it++)
-        {
-          const double & w = it->first;
-          auto constr = it->second;
-          if(!constr->isEquality())
-            assert(false && "Inequalities in the cost function are not implemented yet");
-          
-          EIGEN_MALLOC_ALLOWED;
-          m_H.noalias() += w*constr->matrix().transpose()*constr->matrix();
           EIGEN_MALLOC_NOT_ALLOWED;
-          
-          m_g.noalias() -= w*constr->matrix().transpose()*constr->vector();
-        }
+
+          // Compute the cost 
+          if(problemData.size() > 1)
+          {
+            const ConstraintLevel & cl1 = problemData[1];
+            m_qpData.H.setZero();
+            m_qpData.g.setZero();
         
-        m_H.diagonal().array() += m_hessian_regularization;
-      }
-      
-      STOP_PROFILER_EIQUADPROG_FAST(PROFILE_EIQUADPROG_PREPARATION);
-      
-      
+            for(ConstraintLevel::const_iterator it=cl1.begin(); it!=cl1.end(); it++)
+            {
+              const double & w = it->first;
+              auto constr = it->second;
+              if(!constr->isEquality())
+                assert(false && "Inequalities in the cost function are not implemented yet");
+              
+              EIGEN_MALLOC_ALLOWED;
+              m_qpData.H.noalias() += w*constr->matrix().transpose()*constr->matrix();
+              EIGEN_MALLOC_NOT_ALLOWED;
+              
+              m_qpData.g.noalias() -= w*constr->matrix().transpose()*constr->vector();
+            }
+            
+            if (hessianRegularization)
+            {
+              double m_hessian_regularization(DEFAULT_HESSIAN_REGULARIZATION);
+              m_qpData.H.diagonal().array() += m_hessian_regularization;
+            }
+          }
+    }
```

Comment:
There seems to be an indentation issue here, which makes the changes of the PR hard to understand

Reply:
Oups sorry, I corrected it.